src/Mvc/Mvc.Core/src/Builder/OpenApiEndpointConventionBuilderExtensions.cs

```diff
@@ -120,5 +124,42 @@ public static MinimalActionEndpointConventionBuilder ExcludeFromDescription(this
 
             return Produces<HttpValidationProblemDetails>(builder, statusCode, contentType);
         }
+
+        /// <summary>
+        /// Adds the <see cref="Accepts"/> to <see cref="EndpointBuilder.Metadata"/> for all builders
+        /// produced by <paramref name="builder"/>.
+        /// </summary>
+        /// <typeparam name="TRequest">The type of the request.</typeparam>
+        /// <param name="builder">The <see cref="MinimalActionEndpointConventionBuilder"/>.</param>
+        /// <param name="contentType">The request content type. Defaults to "application/json" if empty.</param>
+        /// <param name="additionalContentTypes">Additional response content types the endpoint produces for the supplied status code.</param>
+        /// <returns>A <see cref="MinimalActionEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static MinimalActionEndpointConventionBuilder Accepts<TRequest>(this MinimalActionEndpointConventionBuilder builder, string contentType, params string[] additionalContentTypes)
+        {
+            Accepts(builder, typeof(TRequest), contentType, additionalContentTypes);
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Adds the <see cref="Accepts"/> to <see cref="EndpointBuilder.Metadata"/> for all builders
+        /// produced by <paramref name="builder"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="MinimalActionEndpointConventionBuilder"/>.</param>
+        /// <param name="requestType">The type of the request. Defaults to null.</param>
+        /// <param name="contentType">The response content type that the endpoint accepts.</param>
+        /// <param name="additionalContentTypes">Additional response content types the endpoint accepts</param>
+        /// <returns>A <see cref="MinimalActionEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static MinimalActionEndpointConventionBuilder Accepts(this MinimalActionEndpointConventionBuilder builder, Type requestType,
+            string contentType, params string[] additionalContentTypes)
+        {
+            if (string.IsNullOrEmpty(contentType))
+            {
+                contentType = "application/json";
+            }
```

Comment:
```suggestion
```

`contentType` is required, so this wouldn't be null (and we generally don't special case empty strings). Let's remove this if block.